faunadb-java/src/main/java/com/faunadb/client/types/SetRef.java

```diff
@@ -0,0 +1,46 @@
+package com.faunadb.client.types;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * A FaunaDB set literal.
+ *
+ * @see <a href="https://faunadb.com/documentation/queries#values-special_types">FaunaDB Special Types</a>
+ */
+@JsonDeserialize(using = Codec.SetRefDeserializer.class)
+public class SetRef extends Value.ScalarValue<ImmutableMap<String, Value>> {
+
+  public SetRef(ImmutableMap<String, Value> parameters) {
+    super(parameters);
+  }
+
+  @JsonValue
+  @Override
+  public ImmutableMap<String, Value> asObject() {
+    return value;
+  }
+
+  @Override
+  public Value get(String key) {
```

Comment:
Should `SetRef` still be a `ScalarValue`? I haven't changed that actually, I think I got form Scala driver.

Perhaps we don't override path navigation methods but, instead, provide special methods only for `SetRef`? I don't know...